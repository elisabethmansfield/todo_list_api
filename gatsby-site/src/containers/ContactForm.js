import React from 'react';
import fetch from 'node-fetch';
import SubHeader from '../components/SubHeader';

export default class ContactForm extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      confirm: undefined,
      name: '',
      message: ''
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  
  handleChange(e) {
    if(this.state.confirm){
      this.setState(() => ({ 
        confirm: undefined
      }));      
    }
    const value = e.target.value.toString();
    const name = e.target.name.toString();
    this.setState({
      [name]: value
    });
    console.log('state: ',this.state.name,this.state.message);
  }

  handleSubmit(e) {
    console.log('Form submitted: ',this.state.name,this.state.message);
    e.preventDefault();
    
    try{
      const body = { 
        name: this.state.name,
        message: this.state.message
      };
      // submit form data with Zapier Webhooks
      fetch(this.props.webhook, {       
        method: 'post',
        headers: {},
        body:    JSON.stringify(body),
      })
      .then(res => res.json())
      .then(json => console.log('Zapier Webhooks: ',json))
      .then(
        this.setState(() => ({ 
          confirm: 'Submitted!'
        }))
      );
    } catch(err){
      return 'Error: ' + err;
    }
  }
  render() {
    return (
      <div>
        <SubHeader title="Contact Form" subtitle="Submit your information" />
        <div className="container">
          <div className="widget">
            <form className="contact-form" action={this.props.webhook} type="POST" onSubmit={this.handleSubmit}>
              <div className="widget-header">
                <h3 className="widget-header__title">Name</h3>
              </div>
              <input className="contact-form__input" type="text" name="name" value={this.state.value} onChange={this.handleChange} />
              <br />
              <div className="widget-header">
                <h3 className="widget-header__title">Message</h3>
              </div>              
              <textarea className="contact-form__input" name="message" value={this.state.value} onChange={this.handleChange} />
              <br />
              <input className="contact-button" type="submit" value="Submit" />
            </form>
            {this.state.confirm && <p className="contact-form__confirm">{this.state.confirm}</p>}
          </div>
        </div>
      </div>
    );
  }
}
