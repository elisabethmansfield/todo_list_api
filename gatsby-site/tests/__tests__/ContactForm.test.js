import React from "react";
import renderer from "react-test-renderer";
import ContactForm from "../../src/containers/ContactForm";

describe("ContactForm", () =>
  it("renders correctly", () => {

    const webhook = '';
    const tree = renderer
      .create(<ContactForm webhook={webhook} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
}));