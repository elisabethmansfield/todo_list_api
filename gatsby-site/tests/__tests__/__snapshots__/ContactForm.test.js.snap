// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm renders correctly 1`] = `
<div>
  <div
    className="header"
  >
    <div
      className="container"
    >
      <h1
        className="header__title"
      >
        Contact Form
      </h1>
      <h2
        className="header__subtitle"
      >
        Submit your information
      </h2>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="widget"
    >
      <form
        action=""
        className="contact-form"
        onSubmit={[Function]}
        type="POST"
      >
        <div
          className="widget-header"
        >
          <h3
            className="widget-header__title"
          >
            Name
          </h3>
        </div>
        <input
          className="contact-form__input"
          name="name"
          onChange={[Function]}
          type="text"
        />
        <br />
        <div
          className="widget-header"
        >
          <h3
            className="widget-header__title"
          >
            Message
          </h3>
        </div>
        <textarea
          className="contact-form__input"
          name="message"
          onChange={[Function]}
        />
        <br />
        <input
          className="contact-button"
          type="submit"
          value="Submit"
        />
      </form>
    </div>
  </div>
</div>
`;
